internal/scheduler/scheduling_algo.go

```diff
@@ -35,6 +35,8 @@ type SchedulingAlgo interface {
 	// Schedule should assign jobs to nodes.
 	// Any jobs that are scheduled should be marked as such in the JobDb using the transaction provided.
 	Schedule(*armadacontext.Context, *jobdb.Txn) (*SchedulerResult, error)
+	// Config returns the configuration used by the scheduling algorithm.
+	Config() *configuration.SchedulingConfig
```

Comment:
I don't think we should pass config back up. If the parent wants the config it should retain it when creating the algo instead of passing it back up again.